datajunction-server/datajunction_server/api/system.py

```diff
@@ -0,0 +1,141 @@
+    Returns a list of DJ system metrics (available as metric nodes in DJ).
+    """
+    settings = get_settings()
+    metrics = await Node.find_by(
+        session=session,
+        namespace=settings.seed_setup.system_namespace,
+        node_types=[NodeType.METRIC],
+    )
+    return [m.name for m in metrics]
+
+
+@router.get("/system/data/{metric_name}")
+async def get_data_for_system_metrics(
+    metric_name: str,
+    dimensions: list[str] = Query([]),
+    filters: list[str] = Query([]),
+    orderby: list[str] = Query([]),
+    session: AsyncSession = Depends(get_session),
+    *,
+    current_user: User = Depends(get_current_user),
+    background_tasks: BackgroundTasks,
+    validate_access: access.ValidateAccessFn = Depends(
+        validate_access,
+    ),
+    cache: Cache = Depends(get_cache),
+) -> list[list[RowOutput]]:
+    """
+    This is not a generic data for metrics endpoint, but rather a specific endpoint for
+    system overview metrics that are automatically defined by DJ, such as the number of nodes.
+    This endpoint will return data for any system metric, cut by their available dimensions
+    and filters.
+
+    This setup circumvents going to the query service to get metric data, since all system
+    metrics can be computed directly from the database.
+
+    For a list of available system metrics, see the `/system/metrics` endpoint. All dimensions
+    for the metric can be discovered through the usual endpoints.
+    """
+    # e.g., "system.dj.number_of_nodes"
+    unique_string = f"{metric_name}:{','.join(dimensions)}:{' AND '.join(filters)}"
+    unique_hash = hashlib.sha256(unique_string.encode()).hexdigest()[:16]
+    cache_key = f"system:sql:{metric_name}:{unique_hash}"
+    if not (translated_sql := cache.get(cache_key)):  # pragma: no cover
+        translated_sql, _ = await get_node_sql(
+            metric_name,
+            dimensions=dimensions,
+            filters=filters,
+            orderby=orderby,
+            session=session,
+            current_user=current_user,
+            background_tasks=background_tasks,
+            validate_access=validate_access,
+        )
+        # A long timeout for the cache is fine here, since these are system nodes whose
+        # definitions should only change upon deployment, at which point the cache can be
+        # retriggered
+        background_tasks.add_task(cache.set, cache_key, translated_sql, timeout=8640000)
+
+    results = await session.execute(text(translated_sql.sql))
+    output = [
+        [
+            RowOutput(
+                value=value,
+                col=col.semantic_entity
+                if col.semantic_type == "dimension"
+                else col.node,
+            )
+            for value, col in zip(row, translated_sql.columns)  # type: ignore
+        ]
+        for row in results
+    ]
+    return output
+
+
+@router.get("/system/dimensions", response_model=list[DimensionStats])
+async def dimensions_stats(
```

Comment:
To match the above naming style:
```suggestion
async def get_dimensions_stats(
```